src/analysisd/testrule.c

```diff
@@ -671,3 +695,72 @@ void onexit() {
 void onsignal(__attribute__((unused)) int signum) {
     exit(EXIT_SUCCESS);
 }
+
+
+void DecodeEventchannelTestRule(Eventinfo *lf)
+{
+    cJSON *event = NULL;
+    cJSON *json_event = NULL;
+    const char *jsonErrPtr;
+    OS_XML xml;
+
+    print_out("\n**Phase 2: Completed decoding.\n");
+    print_out("       decoder: 'windows_eventchannel'");
+
+    /* Obtain event in JSON format */
+    if (!strncmp(lf->log, "{\"win\":", 7))
+    {
+        if (event = cJSON_ParseWithOpts(lf->log, &jsonErrPtr, 0), !event)
+        { print_out("\nMalformed eventchannel JSON event.\n"); }
+
+        if (json_event = cJSON_GetObjectItem(event, "win"), !json_event)
+        { print_out("\nMalformed eventchannel JSON event.\n"); }
+
+        if(printArrayEvent(lf, json_event, "system"))
+        { print_out("\nMalformed JSON: field 'system' not found.\n"); }
+
+        if(printArrayEvent(lf, json_event, "eventdata"))
+        { print_out("\nMalformed JSON: field 'eventdata' not found.\n"); }
+
+        cJSON_Delete(event);
+    }
+    /* Obtain event in XML format */
+    else if (!strncmp(lf->log, "<Event", 6))
+    {
+        if (OS_ReadXMLString(lf->log, &xml) < 0 || DecodeWinevt(lf))
```

Comment:
**OS_ReadXMLString()** is redundant since it's already being used in **DecodeWinevt()**
https://github.com/wazuh/wazuh/blob/187599d519d648f8802a25ecfc005275ee56f115/src/analysisd/decoders/winevtchannel.c#L200-L202